### PR TITLE
Add date format option to DatePickerCell

### PR DESCRIPTION
**Pull request by elvus**
## Change Summary
- packages/nc-gui/components/project/spreadsheet/components/EditColumn.vue
- packages/nc-gui/components/project/spreadsheet/components/EditableCell.vue
- packages/nc-gui/components/project/spreadsheet/components/editColumn/DateOptions.vue (new)
- packages/nc-gui/components/project/spreadsheet/components/editableCell/DatePickerCell.vue
- packages/nc-gui/helpers/dateFormatHelper.js (new)

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

https://user-images.githubusercontent.com/24416449/174604861-9f945b25-6def-4d06-8f87-36c0f93a377f.mp4

### CODE REVIEW

**Comment by wingkwong** [comment]
What date formats would be supported? If there is a predefined list, would it be better to have an auto-complete input field so that users won't input any other invalid formats?

**Comment by elvus** [comment]
Hi @wingkwong I change the text field to autocomplete and add dateformat list, rigth now support this formats:
```
    'DD-MM-YYYY', "MM-DD-YYYY", "YYYY-MM-DD",
    'DD/MM/YYYY', "MM/DD/YYYY", "YYYY/MM/DD",
    'DD MM YYYY', "MM DD YYYY", "YYYY MM DD",
```
You can add more formats in packages/nc-gui/helpers/dateFormatHelper.js

**Review by wingkwong — APPROVED** [review]

**Comment by wingkwong** [comment]
Mind to update the demo video?

**Comment by wingkwong** [comment]
@elvus Please help resolve the conflict.

**Comment by elvus** [comment]
Hi @wingkwong solved!

File: packages/nc-gui/components/project/spreadsheet/components/editColumn/DateOptions.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    :items="dateFormatList"
+    dense
+    outlined
+  />
+</template>
+<script>
+import { dateFormat, validateDateFormat } from '~/helpers/dateFormatHelper'
+export default {
+  name: 'DateOptions',
+  props: ['column', 'meta', 'value'],
```
Comment (dammy001) [inline packages/nc-gui/components/project/spreadsheet/components/editColumn/DateOptions.vue:16]: can we add props type here?

Comment (elvus) [reply packages/nc-gui/components/project/spreadsheet/components/editColumn/DateOptions.vue:16]: Hi @dammy001 sure, what would be the use of this prop?

**Comment by pranavxc** [comment]
Hi @elvus can you do a rebase and push, looks like all ci-cd tests are failing.

**Pull request closed without merging**